tests/suite/nginxgateway_test.go

```diff
@@ -0,0 +1,297 @@
+		if err := verifyNginxGatewayConditions(ng); err != nil {
+			return err
+		}
+
+		condition := ng.Status.Conditions[0]
+
+		if condition.Type != "Valid" {
+			return fmt.Errorf(
+				"expected nginxGateway condition type to be Valid, instead has type %s",
+				condition.Type,
+			)
+		}
+
+		if condition.Reason != "Valid" {
+			return fmt.Errorf("expected nginxGateway reason to be Valid, instead is %s", condition.Reason)
+		}
+
+		if condition.ObservedGeneration != expObservedGen {
+			return fmt.Errorf(
+				"expected nginxGateway observed generation to be %d, instead is %d",
+				expObservedGen,
+				condition.ObservedGeneration,
+			)
+		}
+
+		return nil
+	}
+
+	getNGFPodName := func() (string, error) {
+		podNames, err := framework.GetReadyNGFPodNames(
+			k8sClient,
+			ngfNamespace,
+			releaseName,
+			timeoutConfig.GetTimeout,
+		)
+		if err != nil {
+			return "", err
+		}
+
+		if len(podNames) != 1 {
+			return "", fmt.Errorf("expected 1 pod name, got %d", len(podNames))
+		}
+
+		return podNames[0], nil
+	}
+
+	AfterAll(func() {
+		// re-apply NginxGateway crd to restore NGF instance for following functional tests
+		Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+
+		Eventually(
+			func() bool {
+				ng, err := getNginxGateway(nginxGatewayNsname)
+				if err != nil {
+					return false
+				}
+
+				return verifyNginxGatewayStatus(ng, int64(1)) != nil
+			}).WithTimeout(timeoutConfig.UpdateTimeout).
+			WithPolling(500 * time.Millisecond).
+			Should(BeTrue())
+	})
+
+	When("testing NGF on startup", func() {
+		When("log level is set to debug", func() {
+			It("outputs debug logs and the status is accepted and true", func() {
+				ngfPodName, err := getNGFPodName()
+				Expect(err).ToNot(HaveOccurred())
+
+				ng, err := getNginxGateway(nginxGatewayNsname)
+				Expect(err).ToNot(HaveOccurred())
+
+				gen, err := getNginxGatewayCurrentObservedGeneration(ng)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(verifyNginxGatewayStatus(ng, gen)).To(Succeed())
+
+				Eventually(
+					func() bool {
+						logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+							Container: "nginx-gateway",
+						})
+						if err != nil {
+							return false
+						}
+
+						return strings.Contains(logs, "\"level\":\"debug\"")
+					}).WithTimeout(timeoutConfig.GetTimeout).
+					WithPolling(500 * time.Millisecond).
+					Should(BeTrue())
+			})
+		})
+
+		When("default log level is used", func() {
+			It("only outputs info logs and the status is accepted and true", func() {
+				teardown(releaseName)
+
+				cfg := getDefaultSetupCfg()
+				cfg.debugLogLevel = false
+				setup(cfg)
+
+				ngfPodName, err := getNGFPodName()
+				Expect(err).ToNot(HaveOccurred())
+
+				ng, err := getNginxGateway(nginxGatewayNsname)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(verifyNginxGatewayStatus(ng, int64(1))).To(Succeed())
+
+				Eventually(
```

Comment:
Let's change this to Consistently since we are expecting a log substring to not exist.